File: RECURSION/EASY/Number of Steps to Reduce a Number to Zero/Code2.cpp
/*
PROBLEM LINK:- https://leetcode.com/problems/number-of-steps-to-reduce-a-number-to-zero/
*/

class Solution {
public:
    int numberOfSteps (int num) {
        if(num < 2)
            return num;
        if(num & 1)
            return numberOfSteps(num-1)+1;
        
        return numberOfSteps(num/2)+1;
    }
};

File: RECURSION/EASY/Number of Steps to Reduce a Number to Zero/Code2.py
# PROBLEM LINK:- https://leetcode.com/problems/number-of-steps-to-reduce-a-number-to-zero/

class Solution:
    def numberOfSteps(self, num: int) -> int:
        if num < 2:
            return num
        
        if num & 1:
            return self.numberOfSteps(num-1) + 1
        
        return self.numberOfSteps(num//2) + 1
